DataFormats/PatCandidates/test/testPackedCandidate.cc

```diff
@@ -158,3 +160,99 @@ void testPackedCandidate::testSimulateReadFromRoot() {
   
 }
 
+
+void testPackedCandidate::testPackUnpackTime() {
+  bool debug = false; // turn this on in order to get a printout of the numerical precision you get for the timing in the various encodings
+
+  if (debug) std::cout << std::endl;
+  if (debug) std::cout << "Minimum time error: " << pat::PackedCandidate::unpackTimeError(1) << std::endl;
+  if (debug) std::cout << "Maximum time error: " << pat::PackedCandidate::unpackTimeError(255) << std::endl;
+  float avgres = 0; int navg = 0;
+  for (int i = 2; i < 255; ++i) {
+    float unp = pat::PackedCandidate::unpackTimeError(i);
+    float res = 0.5*(pat::PackedCandidate::unpackTimeError(i+1)-pat::PackedCandidate::unpackTimeError(i-1));
+    avgres += (res/unp); navg++;
+    if (debug) std::cout << " i = " << i << " unp = " << unp << " quant error = " << res << "  packed = " << int(pat::PackedCandidate::packTimeError(unp+0.3*res)) << " and " << int(pat::PackedCandidate::packTimeError(unp-0.3*res)) << std::endl;
+    CPPUNIT_ASSERT(pat::PackedCandidate::packTimeError(unp+0.3*res) == i);
+    CPPUNIT_ASSERT(pat::PackedCandidate::packTimeError(unp-0.3*res) == i);
+  }
+  if (debug) std::cout << "Average rel uncertainty: " << (avgres/navg) << std::endl;
+  if (debug) std::cout << std::endl;
+
+  if (debug) std::cout << "Minimum time standalone (pos): " << pat::PackedCandidate::unpackTimeNoError(+1) << std::endl;
```

Comment:
These could all be behind one if statement... (& same for elsewhere in this file)